### ISSUE TITLE
[bitnami/redis] redis master restarts indefinitely in v17.17.1

### ISSUE THREAD

**Issue by RupakVeerla** [bitnami/charts#18941]
### Name and Version

bitnami/redis v17.17.1

### What architecture are you using?

None

### What steps will reproduce the bug?

Using redis v17.17.1 would result in continous restart of the redis master with following logs

```
INFO 2023-08-29T09:27:46.203433912Z [resource.labels.containerName: redis] 1:C 29 Aug 2023 09:27:46.203 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
INFO 2023-08-29T09:27:46.203495189Z [resource.labels.containerName: redis] 1:C 29 Aug 2023 09:27:46.203 # Redis version=7.0.12, bits=64, commit=00000000, modified=0, pid=1, just started
INFO 2023-08-29T09:27:46.203502641Z [resource.labels.containerName: redis] 1:C 29 Aug 2023 09:27:46.203 # Configuration loaded
INFO 2023-08-29T09:27:46.204559004Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:27:46.204 * monotonic clock: POSIX clock_gettime
INFO 2023-08-29T09:27:46.205752250Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:27:46.205 * Running mode=standalone, port=6379.
INFO 2023-08-29T09:27:46.205792542Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:27:46.205 # Server initialized
INFO 2023-08-29T09:27:46.207119815Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:27:46.206 * Reading RDB base file on AOF loading...
INFO 2023-08-29T09:27:46.207146987Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:27:46.207 # Can't handle RDB format version 11
INFO 2023-08-29T09:27:46.207155811Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:27:46.207 # Error reading the RDB base file appendonly.aof.42.base.rdb, AOF loading aborted
INFO 2023-08-29T09:27:46.207163339Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:27:46.207 # Unrecoverable error reading the append only file appendonly.aof.42.base.rdb: Invalid argument
INFO 2023-08-29T09:32:52.166518364Z [resource.labels.containerName: redis] 1:C 29 Aug 2023 09:32:52.166 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
INFO 2023-08-29T09:32:52.166577917Z [resource.labels.containerName: redis] 1:C 29 Aug 2023 09:32:52.166 # Redis version=7.0.12, bits=64, commit=00000000, modified=0, pid=1, just started
INFO 2023-08-29T09:32:52.166586266Z [resource.labels.containerName: redis] 1:C 29 Aug 2023 09:32:52.166 # Configuration loaded
INFO 2023-08-29T09:32:52.167277028Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:32:52.167 * monotonic clock: POSIX clock_gettime
INFO 2023-08-29T09:32:52.168231431Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:32:52.168 * Running mode=standalone, port=6379.
INFO 2023-08-29T09:32:52.168267889Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:32:52.168 # Server initialized
INFO 2023-08-29T09:32:52.169446943Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:32:52.169 * Reading RDB base file on AOF loading...
INFO 2023-08-29T09:32:52.169482719Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:32:52.169 # Can't handle RDB format version 11
INFO 2023-08-29T09:32:52.169492997Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:32:52.169 # Error reading the RDB base file appendonly.aof.42.base.rdb, AOF loading aborted
INFO 2023-08-29T09:32:52.169501743Z [resource.labels.containerName: redis] 1:M 29 Aug 2023 09:32:52.169 # Unrecoverable error reading the append only file appendonly.aof.42.base.rdb: Invalid argument
INFO 2023-08-29T09:35:41.579416211Z [resource.labels.containerName: redis] 1:C 29 Aug 2023 09:35:41.579 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
```
reverting to v17.16.0 fixes the issue.

### What do you see instead?

![image](https://github.com/bitnami/charts/assets/54211182/7a8fa9c6-b9f3-4cf5-b92a-f87961b08ef7)

**Comment by tsunamaru** [comment]
This issue is caused by upgrading to Redis 7.2 in 17.15.4 and then reverting it back to 7.0 in 17.17.1, which is caused RDB incompatibility:
<img width="609" alt="image" src="https://github.com/bitnami/charts/assets/37580200/0ce1599e-5525-4bf0-ba00-c052f2ef98f1">

The solution is to upgrade Helm chart to 18.x which is using 7.2 by default or revert to 17.17.0

Dear bitnami, I understand you accidentally pushed upgrade but you know is not backward-compatible, and knowing that you reverting it in patch release, where people expect no breaking changes? That's just not acceptable if you promote yourself as a production-grade solution.

Please don't do that again. Cheers.

**Comment by tyriis** [comment]
related to: https://github.com/bitnami/charts/issues/18861

this can be resolved by upgrading to v18 (redis 7.2) you need to manual resolve the stuck pod by deleting it
